tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -806,6 +806,21 @@ class ExecutionTest {
 
     VERIFY_SUCCEEDED(CreateDXGIFactory1(IID_PPV_ARGS(&factory)));
     if (GetTestParamUseWARP(UseWarpByDefault())) {
+
+      // The WARP_DLL runtime parameter can be used to specify a specific DLL to
+      // load.  To force this to be used, we make sure that this DLL is loaded
+      // before attempting to create the device.
+      HMODULE ExplicitlyLoadedWarpDll = NULL;
+      WEX::Common::String WarpDllPath;
+      if (SUCCEEDED(WEX::TestExecution::RuntimeParameters::TryGetValue(
+              L"WARP_DLL", WarpDllPath))) {
+        WEX::Logging::Log::Comment(WEX::Common::String().Format(
+            L"WARP_DLL requested: %ls", (const wchar_t *)WarpDllPath));
```

Comment:
>WarpDllPath [](http://example.com/codeflow?start=57&length=11)

You can call GetBuffer on a WEX::Common::String instead of casting.

Reply:
Yup.  I prefer this though, since it's actually what GetBuffer does implicitly.